packages/snap-preact-components/src/components/Atoms/LoadMore/LoadMore.tsx

```diff
@@ -0,0 +1,299 @@
+								'& .ss__loadMore__progress__indicator__radial__circle__mask__fill': {
+									clip: `rect(0px, ${progressIndicatorWidth! / 2 + 0.5}px, ${progressIndicatorWidth}px, 0px)`,
+									backgroundColor: color || theme?.colors?.primary || '#ccc',
+								},
+								'&.ss__loadMore__progress__indicator__radial__circle__mask--full': {
+									transform: `rotate(${radialAngle}deg)`,
+								},
+							},
+						},
+					},
+				},
+			},
+		});
+	},
+};
+
+export const LoadMore = observer((properties: LoadMoreProps): JSX.Element => {
+	const globalTheme: Theme = useTheme();
+	const defaultProps: Partial<LoadMoreProps> = {
+		loadMoreText: 'Load More',
+		progressIndicator: 'bar',
+		progressIndicatorWidth: properties?.progressIndicator === 'radial' ? 70 : 300,
+		progressIndicatorSize: properties?.progressIndicator === 'radial' ? 10 : 5,
+	};
+
+	const props = mergeProps('loadMore', globalTheme, defaultProps, properties);
+
+	const {
+		pagination,
+		controller,
+		onClick,
+		auto,
+		intersectionOffset,
+		loading,
+		loadMoreText,
+		progressIndicator,
+		hideProgressIndicator,
+		hideProgressText,
+		disableStyles,
+		className,
+		style,
+		styleScript,
+	} = props;
+
+	const subProps: LoadMoreSubProps = {
+		button: {
+			// default props
+			className: 'ss__loadMore__button',
+			// global theme
+			...globalTheme?.components?.button,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+	};
+
+	const store = pagination || controller?.store?.pagination;
+	const isLoading = typeof loading == 'boolean' ? loading : controller?.store?.loading;
+
+	if (!store) {
+		return <Fragment></Fragment>;
+	}
+
+	const styling: { css?: StylingCSS } = {};
+	const stylingProps = { ...props, pagination: store };
+
+	if (styleScript && !disableStyles) {
+		styling.css = [styleScript(stylingProps), style];
+	} else if (!disableStyles) {
+		styling.css = [CSS.LoadMore(stylingProps), style];
+	} else if (style) {
+		styling.css = [style];
+	}
+
+	const autoProps: { ref?: MutableRef<null> } = {};
+	if (auto) {
+		const loadMoreRef = useRef(null);
+		autoProps.ref = loadMoreRef;
+
+		const loadMoreInViewport = useIntersection(loadMoreRef, intersectionOffset || '0px');
+		if (loadMoreInViewport && store.next) {
+			store.next.url.go({ history: 'replace' });
+		}
+	}
+
+	return store.totalResults ? (
+		<CacheProvider>
+			<div
+				{...styling}
+				{...autoProps}
+				className={classnames(
+					'ss__loadMore',
+					`ss__loadMore--${progressIndicator}`,
+					{ 'ss__loadMore--loading': isLoading },
+					{ 'ss__loadMore--auto': auto },
+					className
+				)}
+			>
+				{!auto && (
+					<Button
+						onClick={(e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+							if (onClick && typeof onClick == 'function') {
+								onClick(e);
+							} else if (store.next) {
+								store.next.url.go({ history: 'replace' });
+							}
```

Comment:
is there a reason this onclick was an if else? I feel like normally when provide onClick props they are in addition to the default onclick functionality. For example the onclick prop in facet option components does both 

`onClick={(e: React.MouseEvent<Element, MouseEvent>) => {
							value.url?.link?.onClick(e);
							onClick && onClick(e);
						}}`